src/components/ui/datefield.tsx

```diff
@@ -0,0 +1,72 @@
+import { useTranslation } from "react-i18next";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface DateFieldProps {
+  value?: string;
+  onChange?: (value: string) => void;
+}
+
+export default function DateField({ value = "", onChange }: DateFieldProps) {
+  const [year, month, day] = value.split("-");
+  const { t } = useTranslation();
+
+  const handleDayChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newDay = e.target.value;
+    if (onChange) {
+      onChange(`${year || ""}-${month || ""}-${newDay}`);
+    }
+  };
+
+  const handleMonthChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newMonth = e.target.value;
+    if (onChange) {
+      onChange(`${year || ""}-${newMonth}-${day || ""}`);
+    }
+  };
+
+  const handleYearChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newYear = e.target.value;
+    if (onChange) {
+      onChange(`${newYear}-${month || ""}-${day || ""}`);
+    }
+  };
+
+  return (
+    <div className="flex items-center gap-2">
+      <div className="flex flex-col gap-1">
+        <Label>{t("day")}</Label>
+        <Input
+          type="number"
+          placeholder="DD"
+          value={day}
+          onChange={handleDayChange}
+          data-cy="dob-day-input"
+        />
```

Comment:
can we restrict the individual inputs respective it's possible values?

day - min: 1, max: 31, similarly for month and year too respectively.
```suggestion
        <Input
          type="number"
          placeholder="DD"
          value={day}
          onChange={handleDayChange}
          min={0}
          max={31}
          data-cy="dob-day-input"
        />
```